### PR TITLE
Add Op(aten::_scaled_dot_product_efficient_attention) | feat(torchlib)

### LINKED ISSUE #1160: [torchlib] Implement `aten::_scaled_dot_product_efficient_attention.default`
From models. This op is emitted from **cuda export only**.

        cait_m36_384
        deit_base_distilled_patch16_224
        crossvit_9_240
        mobilevit_s
        pit_b_224
        nanogpt
        timm_vision_transformer
        beit_base_patch16_224
        stable_diffusion_unet

From https://github.com/microsoft/onnx-converters-private/issues/196

cc @justinchuby

### PR DESCRIPTION
**Pull request by titaiwangms**
Fix #1160 
Follow up #1043 It's another `scaled_dot_product_XXX_attention`

It only supports CUDA.

https://github.com/pytorch/pytorch/blob/38ae17d166a001ef6837553d1ddffa111624df27/torch/_meta_registrations.py#L5195-L5236

NOTE: This PR also enables CUDA tests.

### CODE REVIEW

**Comment by titaiwangms** [comment]
@justinchuby I tried to run the test on dev10, but it seems the tests are somehow fixed with CPU. Could you point me the direction to test it with GPU?

**Comment by justinchuby** [comment]
Maybe try changing https://github.com/microsoft/onnxscript/blob/2e22077ef7bfa6d155b1897b76d7bcda676b3080/onnxscript/tests/function_libs/torch_lib/ops_test.py#L164 ? Apparently we assumed cpu everywhere

File: onnxscript/tests/function_libs/torch_lib/extra_opinfo.py [hunk 1]
```diff
@@ -1319,6 +1319,41 @@ def sample_inputs_scaled_dot_product_flash_attention(
...
 
 
+def sample_inputs__scaled_dot_product_efficient_attention(
+    op_info, device, dtype, requires_grad, **kwargs
+):
+    del op_info
+    del kwargs
+
+    make = opinfo_core.partial(
+        opinfo_core.make_tensor, device=device, dtype=dtype, requires_grad=requires_grad
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/extra_opinfo.py:1329]: You may also control device here

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 2]
```diff
@@ -2015,6 +2015,22 @@ def _where_input_wrangler(
         enabled_if=version_utils.torch_older_than("2.1"),
         reason="The operator is not supported in older version.",
     ),
+    TorchLibOpInfo(
+        "ops.aten._scaled_dot_product_efficient_attention",
+        nn_ops.aten__scaled_dot_product_efficient_attention,
+        trace_only=True,
+        tolerance={torch.float32: (3e-4, 1.5e-5)},
+        # Output[0] is OK, but other outputs just have the same shape with zero values
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:2023]: Use the other compare option instead

Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py:2023]: https://github.com/microsoft/onnxscript/blob/d8eb6b58fd5e63d46fdcd4e220ab9e6d991c5de7/onnxscript/tests/function_libs/torch_lib/ops_test_data.py#L1149C16-L1149C16

Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py:2023]: compare_shape_only_for_output

Comment (titaiwangms) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py:2024]: Done

File: onnxscript/tests/function_libs/torch_lib/extra_opinfo.py (hunk 1 above)
Comment (titaiwangms) [reply onnxscript/tests/function_libs/torch_lib/extra_opinfo.py:1333]: I enabled the whole CUDA tests if it's needed by the test.

**Comment by titaiwangms** [comment]
@justinchuby You might want to review again. I changed the function implementation and enabled CUDA tests.

**Pull request merged**